src/js/components/AppListFilterComponent.jsx

```diff
@@ -59,13 +55,22 @@ var AppListFilterComponent = React.createClass({
         filterText: filterText,
         activated: filterText !== "" || state.focused
       });
-      this.props.onChange({[FilterTypes.TEXT]: filterText});
+    }
+  },
+
+  updateFilters: function () {
+    var router = this.context.router;
+    var params = router.getCurrentParams();
+    var query = router.getCurrentQuery();
+
+    if (params != null && query.filterText !== "") {
+      this.context.router.transitionTo("apps", params, query);
     }
   },
 
   handleClearFilterText: function () {
     this.setQueryParam(FilterTypes.TEXT, "");
-    this.props.onChange({[FilterTypes.TEXT]: ""});
+    this.updateFilters();
```

Comment:
I think combined with the line above, this triggers 2 route updates shortly after each other?
You could combine this maybe like this.
```JavaScript
this.updateFilters(FilterTypes.TEXT, "");
```
And get rig of setQueryParam in this case and working directly with the `transitionTo`.